Chatkit/Model/Room.swift

```diff
@@ -0,0 +1,83 @@
+import Foundation
+import CoreData
+
+public struct Room {
+    
+    // MARK: - Properties
+    
+    public let identifier: String
+    public let name: String?
+    public let isPrivate: Bool
+    public let creator: User
+    public let members: Set<User>?
+    public let typingMembers: Set<User>?
+    public let unreadCount: Int64
+    public let lastMessage: Message?
+    public let metadata: Metadata?
+    public let createdAt: Date
+    public let updatedAt: Date
+    public let deletedAt: Date?
+    
+    let objectID: NSManagedObjectID
+    
+    // MARK: - Initializers
+    
+    init(identifier: String, name: String?, isPrivate: Bool, creator: User, members: Set<User>?, typingMembers: Set<User>?, unreadCount: Int64, lastMessage: Message?, metadata: Metadata?, createdAt: Date, updatedAt: Date, deletedAt: Date?, objectID: NSManagedObjectID) {
```

Comment:
We could remove this `init` too.

Reply:
Please see my replay above (related to `User.swft` file).